src/common.jl

```diff
@@ -120,3 +120,17 @@ succprob(d::DiscreteUnivariateDistribution)
 Get the probability of failure.
 """
 failprob(d::DiscreteUnivariateDistribution)
+
+# Temporary fix to handle RFunctions dependencies
+abstract type DiscreteUnivariateRDist <: DiscreteUnivariateDistribution end
```

Comment:
Why is this part needed? I'm not a huge fan of a temporary fix that changes type hierarchies

Reply:
R-based distributions need to be able to behave differently (but in a consistent fashion) to all other distributions to be alternate-RNG aware, as they switch to using a quantile-based generator. In order to pick this up generically using dispatch I introduced the type hierarchy you see. The idea was to minimise the amount of code that needed to be changed to turn them into julia-based distributions, and maximise the visibility of the changes so people would be encouraged to fix them, rather than maintain type-cleanliness.

Reply:
Could you just override the methods for those specific distributions? Could define a simple macro
```julia
macro rand_rdist(D)
    esc(quote
        function rand(d::$D, n::Int)
            [rand(d) for i in Base.OneTo(n)]
        end
        function rand!(d::$D, X::AbstractArray)
            for i in eachindex(X)
                X[i] = rand(d)
            end
            return X
        end
    end)
end
```

Reply:
I guess my point is no one is going to be adding _more_ `DiscreteUnivariateRDist` distributions, so there isn't much to be gained by adding it to the type hierarchy.

Reply:
As I say below, I'm happy to go with this, but I'm not really a macro person so you'll have to tell me exactly what to do :)

Reply:
Okay, I've got the hang of this. Now implemented.